librz/core/cmd_analysis.c

```diff
@@ -975,7 +976,8 @@ static int cmd_an(RzCore *core, bool use_json, const char *name) {
 		if (fcn) {
 			if (name) {
 				ret = rz_analysis_function_rename(fcn, name) ? 0 : -1;
-			} else if (!use_json) {
+			}
+			if (!use_json) {
```

Comment:
Why this?

Reply:
because the code logic is just wrong.
the code is supposed to replace the flag and then print it, but when name is NULL, parts of the code is never executed.